### DISCUSSION TITLE
More HashPrefix  options required

### DISCUSSION THREAD

**Discussion by JackyToU** [juicedata/juicefs#3553] [General]
**What would you like to be added**:

More HashPrefix options required such as  return fmt.Sprintf("%d/%d/%d/%d/%d", sliceId/100/100/100/100/10,  sliceId/100/100/100/10, sliceId/100/100/10, sliceId/100/10, sliceId/10)

**Why is this needed**:
有些存储如果同一目录下文件过多，获取目录下文件列表时会提示“failed get objs: failed to list objs: bad status code 429 Too Many Requests”如果能限制每个目录下的最大文件数量，哪怕多些目录的层次，也就不会出这种问题了。
按照 [文档](https://juicefs.com/docs/community/internals/#data-objects)中的描述，每个slice最多有16个block。这样最底层目录中存储的slice最好不要太多。

**Comment by SandyXSD** [comment]
What object storage you are using when you meet this error?

**Comment by JackyToU** [comment]
Webdav provided by alist converted from aliyundrive阿里云盘

**Comment by zhijian-pro** [comment]
Webdav does not have a limit on the maximum number of files in the directory. Is this a limit made by Ali Cloud disk itself?

**Comment by JackyToU** [comment]
Yes, the limit is by Ali Cloud. 
I guess it is because the amount the records returned by server is limited in each query, so it will take too many requests in a short period of time when trying to fetch the file list the whole directory.
> Webdav does not have a limit on the maximum number of files in the directory. Is this a limit made by Ali Cloud disk itself?

**Comment by davies** [comment]
不打算因为阿里云盘的限制引入这样一个选项（增加维护成本），你可以 fork 一下，在 fork 里面做这个调整。
